apps/dashr-financial-report/assets/FinancialFunctions.R

```diff
@@ -0,0 +1,130 @@
+# Install a package that is necessary and has not been installed, or load one that is.
+
+usePackage <- function(p) 
+{
+  if (!is.element(p, installed.packages()[,1]))
+    install.packages(p, dep = TRUE)
+  require(p, character.only = TRUE)
+}
+
+
+# Generate a Table Function 
+generate_table <- function(df, nrows=20)
+  
+  # function generates a dash table from a supplied data frame (df)
+  
+  #  and number of rows (nrows) to display
+{
+  
+  n <- min(nrows, nrow(df))
+  
+  rows <- lapply(seq(1, n), function(i) {
+    
+    htmlTr(children = lapply(as.character(df[i,]), htmlTd))
+    
+  })
+  
+  header <- htmlTr(children = lapply(names(df), htmlTh))
+  
+  htmlTable(
+    
+    children = c(list(header), rows)
+  )
+}
+
+# Remove.Factors
+
+remove.factors <- function (df) 
+{
+  for (varnum in 1:length(df)) {
+    if ("factor" %in% class(df[, varnum])) {
+      df[varnum] = as.character(df[, varnum])
+    }
+  }
+  return(df)
+}
+
+
+
+#Create a HTML Header:
+
+Header <- function() {
+  return(
+    htmlDiv(list(
+      getLogo(),
+      htmlBr(),
+      getHeader(),
+      htmlDiv(list(
+        grey_line
+      )),
+      htmlBr(list()),
+      getMenu()
+    ))
+  )
+}
+
+
+getLogo <- function() {
+  logo = htmlDiv(list(
+    
+    htmlDiv(list(
+      (htmlImg(src = 'assets/Logo.png', height='40', width='220')
+      )), className = 'ten columns padded'),
+    
+    
+    htmlDiv(list(htmlA( href = "/dashr-financial-report/fullversion",
+                        htmlButton(list("Full Version"), className = "redbutton"
+                        )))),
+    
+    htmlDiv(list(htmlA( href = 'https://github.com/plotly/dash-financial-report',
+                        htmlButton(list("Learn More"), className = "learnbutton"
+                        )))),
+    
+    htmlDiv(list(htmlA(
+      htmlImg(src='https://user-images.githubusercontent.com/1865834/50180824-abcc5f80-02d8-11e9-8319-8842909c3f8e.png', 

```

Comment:
I'd use `assets` for this, store the image locally.

Reply:
Fixed, updated with new Plotly logo.